pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -212,6 +225,59 @@ protected void doShutdown() {
     if (_leaseExtender != null) {
       _leaseExtender.shutDown();
     }
+    // Make sure we do metric cleanup when we shut down the table.
+    _ingestionDelayTracker.shutdown();
```

Comment:
is there any reason we don't shut this down first? I'm testing now, but I feel like this causes lag to artificially go up while a server is shutting down

Reply:
OK what would be the proposal here: where do you suggest we do the shutdown?

Reply:
Sorry, forgot to respond with testing results. I was thinking it should be done at the top of the function before the segment data managers are destroyed. But I tested that, and I still saw lag when servers were shutting down. I've since pivoted to tackling some other ingestion lag issues around losing controllers, but I'll update once I get a chance to come back to this.

Reply:
@jadami10 Once consumption stops, we do tick the measures reported by the metric until the consuming thread is shutdown and yes there may be a ramping delay depending on how long it takes to shutdown things. We can likely improve that by marking shutdown in progress and stopping the ramp in time, this may help with heavily loaded servers to stop ramping during long-lasting shutdown. If that makes sense please file an issue and we can try to reach consensus to adding that feature.

Reply:
filed https://github.com/apache/pinot/issues/11533. I'll try to get our team to chase this down